### PR TITLE
[NCL-7214] Group Config List page

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

**Comment by DnsZhou** [comment]
see jira NCL-7214 for screenshots

**Comment by DnsZhou** [comment]
for some reason, the tests work well on my local but fail on GitHub jobs. Investigation in progress.

File: src/common/groupConfigEntityAttributes.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    title: 'ID',
+  },
+  name: {
+    id: 'name',
+    title: 'Name',
+    filter: {
+      operator: '=like=',
+    },
+  },
+  buildConfigsCount: {
```
Comment (matedo1) [inline src/common/groupConfigEntityAttributes.ts:21]: It's defined, but no used. Let's use it.

File: src/components/GroupConfigsList/GroupConfigsList.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      <ContentBox borderTop>
+        <ServiceContainerLoading {...serviceContainerGroupConfigs} title={PageTitles.groupConfigs}>
+          <TableComposable isStriped variant="compact">
+            <Thead>
+              {/**
+               * If column order is changed, the property tableColumnIndex (see sortAttributes) has to be updated.
+               * Better solution can be implemented in the future.
+               */}
+              <Tr>
+                <Th width={30} sort={getSortParams(sortAttributes['name'].id)}>
```
Comment (matedo1) [inline src/components/GroupConfigsList/GroupConfigsList.tsx:59]: On Product version detail page there is also Group Configs list containing action column. As this component will be reused also there, this column should be added.

Check also other possible places where this component should be reused to make this component reusable also there.

File: src/components/GroupConfigsList/GroupConfigsList.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * @param componentId - Component ID
+ */
+export const GroupConfigsList = ({ serviceContainerGroupConfigs, componentId }: IGroupConfigsList) => {
+  const { getSortParams } = useSorting(sortAttributes, componentId);
+
+  return (
+    <>
+      <Toolbar>
+        <ToolbarItem>
+          <Filtering filterOptions={getFilterAttributes(groupConfigEntityAttributes)} componentId={componentId} />{' '}
```
Comment (matedo1) [inline src/components/GroupConfigsList/GroupConfigsList.tsx:46]: `{' '}` can be removed

File: src/components/GroupConfigsList/GroupConfigsList.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+const sortAttributes: ISortAttributes = {
+  name: {
+    id: 'name',
+    title: 'Name',
+    tableColumnIndex: 0,
+  },
+};
+
+interface IGroupConfigsList {
```
Comment (matedo1) [inline src/components/GroupConfigsList/GroupConfigsList.tsx:28]: `GroupConfigsListProps`

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/common/groupConfigEntityAttributes.ts (hunk 1 above)
Comment (DnsZhou) [reply src/common/groupConfigEntityAttributes.ts:21]: Sorry that I was intended to ask you whether we need this column as well. Will try to implement it.

Comment (matedo1) [reply src/common/groupConfigEntityAttributes.ts:21]: I can see it's still missing?

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/components/GroupConfigsPage/GroupConfigsPage.tsx [hunk 5]
```diff
@@ -1,23 +1,41 @@
...
+  const serviceContainerGroupConfigs = useServiceContainer(groupConfigApi.getGroupConfigs);
+
+  useQueryParamsEffect(serviceContainerGroupConfigs.run, { componentId });
+  useTitle(PageTitles.groupConfigs);
+  return (
+    <PageLayout
+      title={PageTitles.groupConfigs}
+      description={
+        <>
+          This page contains all Group Configs, which are the configurations for group builds. Usually one Group Config may
```
Comment (matedo1) [inline src/components/GroupConfigsPage/GroupConfigsPage.tsx:28]: There is typo mistake - `may contain`.

File: src/common/groupConfigEntityAttributes.ts (hunk 1 above)
Comment (DnsZhou) [reply src/common/groupConfigEntityAttributes.ts:21]: Sorry I forgot to push the commit([747fbda](https://github.com/project-ncl/pnc-web-ui-react/pull/251/commits/747fbdaa68bdc7d224606627f5c8128259d04efd)), pushed accordingly.

Comment (DnsZhou) [reply src/common/groupConfigEntityAttributes.ts:21]: Just found that the commit message was trimmed by mistake, will correct it.

**Comment by DnsZhou** [comment]
Hi @matedo1 , the missing commit was left on my local, and was pushed so it should work now.

**Review by matedo1 — APPROVED** [review]

**Pull request merged**